### DISCUSSION TITLE
Camera in infinite loop after unhack - please help

### DISCUSSION THREAD

**Discussion by lukemikeluke** [roleoroleo/yi-hack-Allwinner-v2#737] [Q&A]
Hi. Could anybode please help me out. 
Camera hacked with y20ga. When tried to unhack it it  goes into this loop. It starts (orange light) connects to wifi, works ok for 10 seconds, and then restarts. Unbrick did not work.
I can log in via webbrowser (for 10 seconds) and disabled snapshot. Removing card does not help. 
I would really appreciate your help. Thank you.

**Comment by roleoroleo** [comment]
You need to apply unbrick procedure.
It's the only way.

**Reply by lukemikeluke** [reply]
Thanks roleoroleo. I did manage to bring it back with an old backup. Thanks for a great hack. The camera works great now but....
Went to hack additional 2 cameras. BFUSY31. Two identical ones. The procedure was: downgrade to 8 firmware with a backup and then hack.
The first one - bang - no problem. The second one - stuck in a loop. Tried the backup - nothing. Restart and tried the hack. It blinked blue and then orange for an hour. Restarted and Right now it's just orange steady light. Is there anything that may be done to save it? Or is the boot loader broken? Thanks in advance. Appreciated.

**Reply by roleoroleo** [reply]
It's not required to downgrade to version 8 before the hack,.
v12 is supported.

About 2nd cam, are you sure it's the same model?
When the led is orange, the bootloader should be ok.
You need tio follow the unbrick procedure, but check if the model is the same.
The backup should contain a file homever,txt with the fw version. This is enough to know the exact model.

**Reply by lukemikeluke** [reply]
Yes 100% sure its BFUSY31. It came from the same box (2pack) when I bought it a while back. I also confirmed the sticker and serial in the app - BFUSY31.
The reason for the downgrade was that IMO the cams work much better with the old firmware (motion detection)
It was odd in the first place. Backup file did not want to work with it for some reason and then this happened. 
The thing is nothing works. I tried building the unbrick files numerous times from 3 different backups, tried hacking it again. There is no reaction, no clicking nothing. It's just orange light when plugged in. No loop nothing. Reset does not work either. Am I f**ed?
Any chance you have a magic bullet, or anything I can put on an SD card to bring it back to life? I'm more than happy to share the backup files too. Thanks a million.

**Reply by roleoroleo** [reply]
If you have a backup of this 2nd, please share it with me.

There is the last chance: a serial connection.
If you have electronic/IT knowledge, I could help you to connect a serial adapter and access to the bootloader.
You need:
- disassemble the cam
- locate the uart pads in the pcb
- buy an usb2uart adapter to connect the usb port of your pc to the uart of the cam
- a lot of patience

There is a guide for another model, if you want to check it:
https://github.com/roleoroleo/yi-hack-Allwinner/wiki/Dump-your-backup-firmware-(serial-cable)
https://github.com/roleoroleo/yi-hack-Allwinner/issues/365

**Reply by lukemikeluke** [reply]
I shared a repo with the backup. Hope this way's ok. Thanks.

**Reply by roleoroleo** [reply]
The cam is a y20ga
I got confused because you opened the issue in the wrong repo.
Check the repo for Allwinner and not Allwinner-v2
https://github.com/roleoroleo/yi-hack-Allwinner
Which unbrick procedure did you used?

**Reply by lukemikeluke** [reply]
Sorry for the confusion. This is the exact unbrick procedure I used. https://github.com/roleoroleo/yi-hack-Allwinner/wiki/Unbrick-the-cam
Cloned the repo -> copy the backup to y20ga -> run the built command -> unpack the 2 files into the cam SD. Worked well on identical cam. 
On this one does not. Nothing. Just orange static light :( no clicking :(. I added the recovery files I created to the repo. Thanks Man.

**Reply by roleoroleo** [reply]
Files are ok.
The last chance is the serial connection.

**Reply by lukemikeluke** [reply]
Oh no :(. Anyway if I decide to do it, I will take you on you offer (but would have to buy usb adapter first, soldering would be secondary to that). Thank you very much for your help and great hacks.

**Reply by roleoroleo** [reply]
Another question please: which option did you use to create the image?
hacked or factory?

**Reply by lukemikeluke** [reply]
Factory. Is that the right way to go?

**Reply by roleoroleo** [reply]
There isn't a right way. It depends...
Try hacked.
There is a loop prevention mechanism: the bootloader doesn't load the image if the hash is the same as the current one.
So you can try 1000 times but it will not work.
In the "hacked" version I added a random file to force the hash to change.

**Reply by lukemikeluke** [reply]
Oh Man you raised my hopes up. So I build the hacked version with the backup -> copied onto the sd card -> crossed my fingers and -> darn ... it didn't work. Cursed orange light still there. Thank for your help Man. I think it's dead :(.

**Reply by roleoroleo** [reply]
If the led is orange, the cam is alive.
If your cam is based on fw 8, try this unbrick file:
[home_y20ga.gz](https://github.com/roleoroleo/yi-hack-Allwinner-v2/files/13328669/home_y20ga.gz)

**Reply by lukemikeluke** [reply]
There is a progress, well kind off. After using your file the camera blinked once and clinked once, and now it's blinking orange!
Full of hope I tried my backup (both factory and hacked option).
If I use my hacked option built backup -> it's back to steady orange
If I use my factory option built backup -> it starts looping - blinking click blinking click .......
If i use your backup -> it's back to blinking without cliking and looping after one click. If I press reset button on it it's back to steady orange.

Before I tried to downgrade to fw 8.0 it it was running 12. Any chance you may have some fw12 magic backup?
Also maybe I try making backup files off of running hacked camera and build a backup  on these and try to run it with it. Is it a good idea? I can't stop now, there is progress !!! F*ck yeah!

**Reply by roleoroleo** [reply]
I will send a v12 backup.

**Reply by roleoroleo** [reply]
[home_y20ga.gz](https://github.com/roleoroleo/yi-hack-Allwinner-v2/files/13328878/home_y20ga.gz)
[rootfs_y20ga.gz](https://github.com/roleoroleo/yi-hack-Allwinner-v2/files/13328880/rootfs_y20ga.gz)

**Reply by lukemikeluke** [reply]
Thank you for the 12 backup. I tried it straight away. But it's the same story as with the 8 backup. Steady light, click, and then blinking orange. So I tried different options last night:
1. Backed up hacked 8.0 camera - > build factory/hacked - > steady - double click - blinking
2.  Tried different SD card, as I thought it may be a reason - > didn't help
3. Tried deferent configurators of your backup and mine -> no.
I can't put my finger on why it's not unbricking. I mean we are doing everything by the book. When It was steady orange I was thinking it makes no sense but now when it's looping or blinking it should come back to life.  
But for some reason it doesn't. It's like I am missing something. Damn it !
I must have done something unique to it when I initially tried to hack it after it bricked due to a firmware downgrade. 
Later today I'll try some more combinations. Fingers crossed.

**Reply by lukemikeluke** [reply]
HI there. Just a quick update. Tried different combinations with similar results, i.e. orange blinking loop and static orange after hard reset. Also tried the unhack from here: https://github.com/TheCrypt0/yi-hack-v4  that gave a short blue blinking light. 
Anyway, will have to give up for now, maybe one day something else will pop up which will bring it back to life.  
Will order a replacement (dome x) which hopefully is going to be hackable.  Thank you again, for your hack (which works great on the other camera) and your willingness to help out. All the best.
